SignalServiceKit/src/Contacts/TSThread.m

```diff
@@ -714,6 +714,21 @@ - (BOOL)hasPendingFriendRequest
         || self.friendRequestStatus == TSThreadFriendRequestStatusRequestReceived;
 }
 
+- (BOOL)isContactFriend
+{
+    return self.friendRequestStatus == TSThreadFriendRequestStatusFriends;
+}
+
+- (BOOL)hasCurrentUserSentFriendRequest
+{
+    return self.friendRequestStatus == TSThreadFriendRequestStatusRequestSent;
```

Comment:
You would also need to check for the expired state here

Reply:
Yeah, didn't do expiration yet